Source/DafnyCore/Resolver/NameResolutionAndTypeInference.cs

```diff
@@ -336,7 +336,9 @@ public partial class Resolver {
       if (expr is ParensExpression) {
         var e = (ParensExpression)expr;
         ResolveExpression(e.E, resolutionContext);
-        e.ResolvedExpression = e.E;
+        var innerRange = e.E.RangeToken;
+        e.ResolvedExpression = e.E; // Overwrites the range, which is not suitable for ParensExpressions
```

Comment:
I also found a post-resolution bug. Normally, when ConcreteExpressions are resolved, their replacement takes the range token of the original expression.
However, when writing `(E)`, during resolution, it would transform the range of `E` to actually contain the parentheses, which would mess up token ownership and prevent formatting from happening correctly post-resolution.